### DISCUSSION TITLE
A quick update from the governance group

### DISCUSSION THREAD

**Discussion by thoughtpolice** [jj-vcs/jj#4667] [General]
Hi folks,

I just wanted to give a quick update on what you're going to see next from the governance group. It's pretty simple: we're currently starting to draft a `GOVERNANCE.md` document that we'll be putting up for review, which will define some of the rules and roles of contributors and maintainers, and which we hope will start as the foundation for a formal governance structure.

This goal of a document like this is to cover key basic ideas such as (but not limited to):

- How do disagreements get handled?
- Defining a contributor role and maintainer role, the responsibilities of each role, and the criteria to become one
- Clarifying that maintainers manage and decide on contributions, but the details will come in a later document

One of the reasons we originally formed the group was the belief that more formal governance is going to be needed as the project grows. Currently, Martin tries to make decisions by consensus, but is _perceived_ as an effective BDFL for the project which is something we want to move away from.

So to start, this basic model will help us design processes around code and design review. We want to make it clear that contribution isn’t becoming limited, but rather will have actual rules written around who can give final approvals or how disagreements can be handled, and how the burden of responsibility is shared.

In more detail, we're going to be basing our initial drafts of the governance rules on some prior art we've been keeping an eye on. After a lot of looking around, we settled on using the [Cloud Native Computing Foundation][CNCF]'s [guidelines for governance][CNCF-gov], and projects underneath their umbrella, as a starting point. You've probably heard of some of those projects &mdash; including Vitess, linkerd, and CoreDNS.

[CNCF]: https://cncf.io
[CNCF-gov]: https://contribute.cncf.io/maintainers/governance/overview/

While the CNCF has overarching guidelines, each project has unique rules on how voting for things works, how maintainers are nominated or elected, and how technical decisions get made. We'll be plucking some ideas from each of these in order to get a draft rolling. Once we do, we'll open a pull request adding a `GOVERNANCE.md` to the repository for review (Stage 2 of the [the temporary governance process][pr-4440]) to get feedback from the community and core maintainers.

[pr-4440]: https://github.com/martinvonz/jj/pull/4400

Just wanted to give everyone a heads up. (If you have any pointers for *other* projects we should look at, let me know!) Take care!

P.S. In case you missed it, you may want to check out the results of the recent Community Survey we put out: <https://github.com/martinvonz/jj/discussions/4577>.

**Comment by mahyarmirrashed** [comment]
This is great, wishing you great success!
